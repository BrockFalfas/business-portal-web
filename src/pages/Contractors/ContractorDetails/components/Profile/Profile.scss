@import "../../../../../styles/var";

.Profile {
  &--button {
    margin-left: 0.6rem;
  }
  &-link {
    color: $primary-color;
    text-decoration: underline;
    &:hover{
      cursor: pointer;
    }
  }
  &-box-informations {
    display: flex;
    justify-content: space-between;
    align-items: center;

    margin-top: 2rem;
    margin-bottom: 4rem;
  }

  &-name {
    font-size: 1.725rem;
    color: black;
  }

  &-address,
  &-phone {
    display: flex;

    .Profile-label {
      font-style: italic;
      margin-right: 1rem;
    }

    .Profile-value {
      font-weight: 500;
      color: black;
    }
  }

  &-filters {
    display: flex;
    align-items: center;
    justify-content: center;
  }

  &-period {
    font-size: 1.25rem;
  }

  &-icon{
    margin-left: 1rem;
  }

  &-warning{
    color: $color-error;
  }
}
