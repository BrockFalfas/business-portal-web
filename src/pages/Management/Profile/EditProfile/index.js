import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Button } from 'antd';
import { Formik } from 'formik';
import classnames from 'classnames';
import _ from 'lodash';

import FormField from '~components/FormField';
import { initialValues, formFields, validationSchema } from './formSchema';
import NotificationService from '~services/notification';
import { handleFormHttpResponse } from '~utils/forms/errors';
import './EditProfile.scss';

export class EditProfile extends React.Component {
  static propTypes = {
    user: PropTypes.object,
    updateProfile: PropTypes.func.isRequired,
    disabled: PropTypes.bool,
  };

  state = {
    error: null,
  };

  render() {
    const { user } = this.props;
    let formFields = user;
    if (formFields === null) {
      formFields = initialValues;
    }
    _.forIn(formFields, (value, key) => {
      if (value === '' || !value) {
        delete formFields[key];
      }
    });
    return (
      <div className="EditProfile">
        <Formik
          initialValues={formFields}
          onSubmit={this.handleSubmit}
          validationSchema={validationSchema}>
          {this.renderForm}
        </Formik>
      </div>
    );
  }

  renderForm = ({ handleSubmit, isSubmitting, values, dirty }) => (
    <form onSubmit={handleSubmit}>
      {Object.entries(formFields).map(([name, options]) => (
        <FormField
          key={name}
          name={name}
          label={options.label}
          disabled={this.props.disabled}
          {...options.input}
          placeholder={this.props.disabled ? '' : options.input.placeholder}
          className={classnames('EditCompanyDetails__input', {
            'EditCompanyDetails__input--disabled': this.props.disabled,
          })}
        />
      ))}

      {!this.props.disabled && (
        <div className="EditCompanyDetails__button-container">
          <Button
            disabled={!dirty || isSubmitting}
            size="large"
            type="primary"
            loading={isSubmitting}
            htmlType="submit"
            className="EditCompanyDetails__button-container--button">
            Save
          </Button>
        </div>
      )}
    </form>
  );

  handleSubmit = async (data, form) => {
    const { updateProfile } = this.props;
    let profileData = validationSchema.cast(data);
    try {
      await updateProfile(profileData);
      this.handleSubmitSuccess();
    } catch (err) {
      handleFormHttpResponse(form, err.response.data.error, err.response);
    }
  };

  handleSubmitSuccess = () => {
    const { history } = this.props;
    this.setState({ error: null });

    NotificationService.open({
      type: 'success',
      message: 'Success',
      description: 'Profile successfully updated.',
    });
    history.push(`/management/profile`);
  };
}

const mapStateToProps = state => ({
  user: state.auth.user,
});

const mapDispatchToProps = ({ users: { updateProfile } }) => ({
  updateProfile,
});

export default connect(mapStateToProps, mapDispatchToProps)(EditProfile);
