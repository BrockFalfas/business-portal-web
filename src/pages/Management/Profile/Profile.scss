@import "../../../styles/var";

.Profile {
  padding: 1rem;
  width: 100%;
  &__button-container {
    width: 100%;
    text-align: center;
    margin-top: 4rem;
  &--button {
      padding: 1rem 3rem;
    }
  }
  &__errors {
    color: $color-inactive;
  }

  .InputGroup {
    width: 50%;
  }

  .ant-calendar-picker {
    width: 100%;
  }
  &__spinner {
    display: block !important;
    margin: 0 auto !important;
  }
  .Box {
    border-radius: 0.9rem 0.9rem 0 0;
  }
  &__box {
    padding: 1rem;
    min-height: 140px;
  }
  &--icon{
    margin-left: 0.2rem;
  }
  &--warning{
    color: $color-error;
  }
  &--link {
    color: $primary-color;
    text-decoration: underline;
    &:hover{
      cursor: pointer;
    }
  }
}
