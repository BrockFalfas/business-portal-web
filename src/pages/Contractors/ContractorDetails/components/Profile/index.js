import React from 'react';

import PropTypes from 'prop-types';
import { Button, Icon, Popover } from 'antd';
import Activity from '../Activity';

import { renderDate } from '~utils/time';
import { formatPhone } from '~utils/number';

import './Profile.scss';
import RefreshButton from '~components/RefreshButton';

export default class Profile extends React.PureComponent {
  static propTypes = {
    handleRefresh: PropTypes.func,
    isLoading: PropTypes.bool,
    hasFundingSource: PropTypes.bool,
    openAddFundingSourceModal: PropTypes.func,
    handleGoToFundingSources: PropTypes.func,
  };

  render() {
    const {
      firstName,
      lastName,
      updatedAt,
      createdAt,
      address1,
      address2,
      city,
      state,
      postalCode,
      phone,
      children,
      handleRefresh,
      isLoading,
      handleGoToFundingSources,
    } = this.props;
    const warningsList = this.verifyUserProfile();
    return (
      <div className="Profile">
        <div className="Profile-box-informations">
          <div className="Profile-basic-data">
            <div className="Profile-name">
              {firstName} {lastName}
              {warningsList.length > 0 && (
                <Popover content={this.renderPopOver(warningsList)} title="Warning">
                  <Icon type="exclamation-circle" theme="twoTone" className="Profile-icon" />
                </Popover>
              )}
            </div>
            <div className="Profile-since">Contractor since {renderDate(createdAt)}</div>
          </div>
          <div className="Profile-activity">
            <Button
              type="primary"
              ghost
              className="Profile--button"
              onClick={handleGoToFundingSources}>
              Funding Sources
            </Button>
            <Activity lastActivityDate={updatedAt} />
            <RefreshButton handleRefresh={handleRefresh} isLoading={isLoading} />
          </div>
        </div>
        <div className="Profile-box-informations">
          <div className="Profile-address">
            <div className="Profile-label">Address</div>
            <div className="Profile-value">
              <div>
                {address1} {address2}
              </div>
              <div>
                {city} {state}
                {postalCode}
              </div>
            </div>
          </div>
          <div className="Profile-phone">
            <div className="Profile-label">Phone</div>
            <div className="Profile-value">{formatPhone(phone)}</div>
          </div>
          <div className="Profile-options">{children}</div>
        </div>
      </div>
    );
  }

  verifyUserProfile = () => {
    const { hasFundingSource } = this.props;
    const warnings = [];
    if (!hasFundingSource) {
      const { openAddFundingSourceModal } = this.props;
      warnings.push({
        key: 'Funding Source',
        content: (
          <p className="Profile-warning">
            1. Add funding source for this user{' '}
            <span className="Profile-link" onClick={openAddFundingSourceModal}>
              here
            </span>.
          </p>
        ),
      });
    }
    return warnings;
  };

  renderPopOver = warnings => {
    return <div>{warnings.map(warning => <div key={warning.key}>{warning.content}</div>)}</div>;
  };
}
