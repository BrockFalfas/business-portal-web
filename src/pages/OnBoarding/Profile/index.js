import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Button } from 'antd';
import { Formik } from 'formik';

import FormField from '~components/FormField';
import { initialValues, formFields, transformDateToMoment, validationSchema } from './formSchema';
import { handleFormHttpResponse } from '~utils/forms/errors';
import { traverseRecursively } from '~utils/iterators';
import './Profile.scss';

export class Profile extends React.Component {
  static propTypes = {
    user: PropTypes.object,
  };

  state = {
    error: null,
  };

  render() {
    const { error } = this.state;
    return (
      <div className="Profile">
        <div className="Profile__form">
          <div className="Profile__errors">{error}</div>
          <Formik
            initialValues={initialValues}
            onSubmit={this.handleSubmit}
            validationSchema={validationSchema}>
            {this.renderForm}
          </Formik>
        </div>
      </div>
    );
  }

  prepareForm(fields) {
    let formArray = [];
    traverseRecursively(fields, {
      childKey: 'fields',
      nodeCallback: () => console.log(),
      leafCallback: data => {
        const { value, path } = data;
        formArray.push(
          <FormField
            key={path.join('.')}
            name={path.join('.')}
            label={value.label}
            {...value.input}
          />
        );
      },
    });
    return [...formArray];
  }

  renderForm = ({ handleSubmit, isSubmitting, values, dirty }) => (
    <form onSubmit={handleSubmit}>
      {this.prepareForm(formFields)}

      <div className="Profile__button-container">
        <Button
          disabled={!dirty || isSubmitting}
          size="large"
          type="primary"
          loading={isSubmitting}
          htmlType="submit"
          className="Profile__button-container--button">
          Next
        </Button>
      </div>
    </form>
  );

  create = async data => {
    const { createContractor, user } = this.props;
    let dataProfile = JSON.parse(JSON.stringify(data));
    dataProfile.profile.dateOfBirth = transformDateToMoment(dataProfile.profile.dateOfBirth).format(
      'YYYY-MM-DD'
    );
    dataProfile.profile.postalCode = String(dataProfile.profile.postalCode);
    dataProfile.profile.email = user.email;

    await createContractor(dataProfile);
  };

  handleSubmit = async (data, form) => {
    const validData = validationSchema.cast(data);
    validData.profile['country'] = 'US';

    try {
      await this.create(validData);

      this.handleSubmitSuccess();
    } catch (err) {
      handleFormHttpResponse(form, err.response.data.error, err.response);
    }
  };

  handleSubmitSuccess = () => {
    const { changeStep } = this.props;
    changeStep(2);
  };
}

const mapStateToProps = state => ({
  user: state.auth.user,
  isLoading: state.loading.effects.onBoarding.createContractor,
});

const mapDispatchToProps = dispatch => ({
  createContractor: dispatch.onBoarding.createContractor,
  changeStep: dispatch.onBoarding.changeStep,
});

export default connect(mapStateToProps, mapDispatchToProps)(Profile);
