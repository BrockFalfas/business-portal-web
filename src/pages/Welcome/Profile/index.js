import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Button } from 'antd';
import { Formik } from 'formik';
import classnames from 'classnames';

import FormField from '~components/FormField';
import { initialValues, formFields, validationSchema } from './formSchema';
import { handleFormHttpResponse } from '~utils/forms/errors';
import './Profile.scss';

export class Profile extends React.Component {
  static propTypes = {
    updateProfile: PropTypes.func.isRequired,
    disabled: PropTypes.bool,
  };

  state = {
    error: null,
  };

  render() {
    return (
      <div className="EditProfile">
        <Formik
          initialValues={initialValues}
          onSubmit={this.handleSubmit}
          validationSchema={validationSchema}>
          {this.renderForm}
        </Formik>
      </div>
    );
  }

  renderForm = ({ handleSubmit, isSubmitting, values, dirty }) => (
    <form onSubmit={handleSubmit}>
      {Object.entries(formFields).map(([name, options]) => (
        <FormField
          key={name}
          name={name}
          label={options.label}
          disabled={this.props.disabled}
          {...options.input}
          placeholder={this.props.disabled ? '' : options.input.placeholder}
          className={classnames('EditCompanyDetails__input', {
            'EditCompanyDetails__input--disabled': this.props.disabled,
          })}
        />
      ))}

      {!this.props.disabled && (
        <div className="EditCompanyDetails__button-container">
          <Button
            disabled={!dirty || isSubmitting}
            size="large"
            type="primary"
            loading={isSubmitting}
            htmlType="submit"
            className="EditCompanyDetails__button-container--button">
            Next
          </Button>
        </div>
      )}
    </form>
  );

  handleSubmit = async (data, form) => {
    const { updateProfile } = this.props;
    let profileData = validationSchema.cast(data);
    try {
      await updateProfile(profileData);
      this.handleSubmitSuccess();
    } catch (err) {
      handleFormHttpResponse(form, err.response.data.error, err.response);
    }
  };

  handleSubmitSuccess = () => {
    const { changeStep } = this.props;
    changeStep(2);
  };
}

const mapStateToProps = state => ({
  isLoading: state.loading.effects.users.updateProfile,
});

const mapDispatchToProps = dispatch => ({
  updateProfile: dispatch.users.updateProfile,
  changeStep: dispatch.onBoarding.changeStep,
});

export default connect(mapStateToProps, mapDispatchToProps)(Profile);
