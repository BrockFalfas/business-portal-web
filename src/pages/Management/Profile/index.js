import React from 'react';
import { Icon } from 'antd';

import Box from '~components/Box';
import Header from '~components/Header';
import TooltipButton from '~components/TooltipButton';
import './Profile.scss';
import EditProfile from './EditProfile';

export class Profile extends React.Component {
  handleEdit = () => {
    this.props.history.push(`/management/profile/edit`);
  };

  handleChangePassword = () => {
    this.props.history.push(`/management/change-admin-password`);
  };

  render() {
    return (
      <div className="Profile">
        <Header title="Profile" size="medium">
          <TooltipButton tooltip="Edit profile" type="primary" onClick={this.handleEdit}>
            <Icon type="form" theme="outlined" />
          </TooltipButton>
          <TooltipButton
            tooltip="Change password"
            type="primary"
            onClick={this.handleChangePassword}>
            <Icon type="lock" theme="outlined" />
          </TooltipButton>
        </Header>
        <Box className="Profile__box">
          <EditProfile disabled />
        </Box>
      </div>
    );
  }
}

export default Profile;
