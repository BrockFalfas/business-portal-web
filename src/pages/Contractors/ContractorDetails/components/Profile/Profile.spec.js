import React from 'react';
import { shallow } from 'enzyme';

import Profile from './index';

const initProfile = overrides => {
  const mockProps = {
    handleRefresh: jest.fn(),
    openAddFundingSourceModal: jest.fn(),
    isLoading: false,
    firstName: 'John',
    lastName: 'Doe',
    updatedAt: '',
    createdAt: '',
    address1: 'Address1',
    address2: 'Address2',
    city: 'San Francisco',
    state: 'CA',
    postalCode: '74234',
    phone: '5551234567',
    children: '',
    accountNumber: '123123123',
    accountRouting: '345345345',
  };

  const wrapper = shallow(<Profile {...mockProps} {...overrides} />);
  return { mockProps, wrapper };
};

describe('component: <Profile/>', () => {
  describe('Profile', () => {
    it('should render without crashing', () => {
      const { wrapper } = initProfile();

      expect(wrapper).toBeTruthy();
    });
  });

  it('render Popover if any warnings', () => {
    const { wrapper } = initProfile();

    expect(wrapper).toMatchSnapshot();
  });
});
