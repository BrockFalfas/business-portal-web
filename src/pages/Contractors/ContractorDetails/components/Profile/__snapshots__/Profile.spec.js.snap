// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: <Profile/> render Popover if any warnings 1`] = `
<div
  className="Profile"
>
  <div
    className="Profile-box-informations"
  >
    <div
      className="Profile-basic-data"
    >
      <div
        className="Profile-name"
      >
        John
         
        Doe
      </div>
      <div
        className="Profile-since"
      >
        Contractor since 
        Invalid Date
      </div>
    </div>
    <div
      className="Profile-activity"
    >
      <Activity
        lastActivityDate=""
      />
      <RefreshButton
        handleRefresh={[Function]}
        isLoading={false}
        size="default"
        type="primary"
      />
    </div>
  </div>
  <div
    className="Profile-box-informations"
  >
    <div
      className="Profile-address"
    >
      <div
        className="Profile-label"
      >
        Address
      </div>
      <div
        className="Profile-value"
      >
        <div>
          Address1
           
          Address2
        </div>
        <div>
          San Francisco
           
          CA
          74234
        </div>
      </div>
    </div>
    <div
      className="Profile-phone"
    >
      <div
        className="Profile-label"
      >
        Phone
      </div>
      <div
        className="Profile-value"
      >
        (555) 123-4567
      </div>
    </div>
    <div
      className="Profile-options"
    />
  </div>
</div>
`;
